### ISSUE TITLE
libproxy version 0.4.16 installation failure

### ISSUE THREAD

**Issue by byxrolland** [Homebrew/linuxbrew-core#21749]
When try to install libproxy with the latest version 0.4.16, it failed.

```
[ 12%] Building CXX object libproxy/CMakeFiles/pxgsettings.dir/modules/pxgsettings.cpp.o
cd /tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build/libproxy && /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/g++-5 -D_POSIX_C_SOURCE=200112L -I/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16 -I/home/linuxbrew/.linuxbrew/Cellar/dbus/1.12.20/include/dbus-1.0 -I/home/linuxbrew/.linuxbrew/Cellar/dbus/1.12.20/lib/dbus-1.0/include -I/home/linuxbrew/.linuxbrew/Cellar/libffi/3.3/include -I/home/linuxbrew/.linuxbrew/Cellar/zlib/1.2.11/include -I/home/linuxbrew/.linuxbrew/Cellar/pcre/8.44/include -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.36/include/libmount -I/home/linuxbrew/.linuxbrew/Cellar/util-linux/2.36/include/blkid -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.66.2_1/include -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.66.2_1/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/Cellar/glib/2.66.2_1/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/opt/gettext/include -fvisibility=hidden  -DNDEBUG -o CMakeFiles/pxgsettings.dir/modules/pxgsettings.cpp.o -c /tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/modules/pxgsettings.cpp
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp: In function 'std::__cxx11::string recvline(int)':
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:37: error: 'errno' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                     ^
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:46: error: 'EINTR' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                              ^
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp: In function 'std::__cxx11::string recvline(int)':
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:37: error: 'errno' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                     ^
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:46: error: 'EINTR' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                              ^
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp: In function 'std::__cxx11::string recvline(int)':
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:37: error: 'errno' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                     ^
/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/libproxy/url.cpp:403:46: error: 'EINTR' was not declared in this scope
  } while (ret == 1 || (ret == -1 && errno == EINTR));
                                              ^
make[2]: *** [libproxy/test/CMakeFiles/get-pac-test.dir/build.make:98: libproxy/test/CMakeFiles/get-pac-test.dir/__/url.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [libproxy/test/CMakeFiles/url-test.dir/build.make:98: libproxy/test/CMakeFiles/url-test.dir/__/url.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [libproxy/test/CMakeFiles/url-encode.dir/build.make:98: libproxy/test/CMakeFiles/url-encode.dir/__/url.cpp.o] Error 1
make[2]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
make[1]: *** [CMakeFiles/Makefile2:1655: libproxy/test/CMakeFiles/url-encode.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
make[1]: *** [CMakeFiles/Makefile2:1628: libproxy/test/CMakeFiles/get-pac-test.dir/all] Error 2
[ 14%] Linking CXX executable pxgsettings
cd /tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build/libproxy && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.19.1/bin/cmake -E cmake_link_script CMakeFiles/pxgsettings.dir/link.txt --verbose=1
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/g++-5 -fvisibility=hidden  -DNDEBUG -rdynamic CMakeFiles/pxgsettings.dir/modules/pxgsettings.cpp.o -o pxgsettings   -L/home/linuxbrew/.linuxbrew/Cellar/dbus/1.12.20/lib  -L/home/linuxbrew/.linuxbrew/Cellar/glib/2.66.2_1/lib  -L/home/linuxbrew/.linuxbrew/opt/gettext/lib  -Wl,-rpath,/home/linuxbrew/.linuxbrew/Cellar/dbus/1.12.20/lib:/home/linuxbrew/.linuxbrew/Cellar/glib/2.66.2_1/lib:/home/linuxbrew/.linuxbrew/opt/gettext/lib: -lgio-2.0 -lgobject-2.0 -lglib-2.0
make[2]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
[ 14%] Built target pxgsettings
make[2]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
make[1]: *** [CMakeFiles/Makefile2:1682: libproxy/test/CMakeFiles/url-test.dir/all] Error 2
[ 15%] Linking CXX static library libmodman.a
cd /tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build/libmodman && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.19.1/bin/cmake -P CMakeFiles/modman.dir/cmake_clean_target.cmake
cd /tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build/libmodman && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.19.1/bin/cmake -E cmake_link_script CMakeFiles/modman.dir/link.txt --verbose=1
/home/linuxbrew/.linuxbrew/bin/ar qc libmodman.a CMakeFiles/modman.dir/module_manager.cpp.o
/home/linuxbrew/.linuxbrew/bin/ranlib libmodman.a
make[2]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
[ 15%] Built target modman
make[1]: Leaving directory '/tmp/libproxy-20201210-6961-gcz11c/libproxy-0.4.16/build'
make: *** [Makefile:185: all] Error 2
```

Output of brew config

```
HOMEBREW_VERSION: 2.6.1
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 2c77a540b522c2eee74af0745851167412adb83b
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core.git
Core tap HEAD: d9e5d7448209ab737c30bb76c54db15e790b2dbf
Core tap last commit: 13 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: 172.20.96.1:0.0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit ivybridge
Clang: 11.0 build (parse error)
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /usr/bin/curl
Kernel: Linux 4.19.104-microsoft-standard x86_64 GNU/Linux
OS: Ubuntu 20.04.1 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: N/A
gcc: 5.5.0_7
xorg: 20170115_1
```
And it was run on WSL2.

**Comment by danielnachun** [comment]
Libproxy 0.4.16 needs a newer version of gcc than gcc 5 (the default).  A PR to fix this needs to made (I will get to it this weekend if nobody else can) as other packages are now failing CI because of this.  If you need this urgently, you can install a newer version of gcc with `brew install gcc@10` for example, and install `libproxy` with `brew install --cc=gcc-10` to force it to use gcc 10 instead of gcc 5.

**Comment by byxrolland** [comment]
@danielnachun I tried to use gcc-10 to install before. But it seems linuxbrew lacks the appropriate support of dbus library.
```
[ 95%] Built target proxy
/tmp/libproxy-20201211-2743-1b34n46/libproxy-0.4.16/libproxy/modules/config_pacrunner.cpp:24:10: fatal error: dbus/dbus.h: No such file or directory
   24 | #include <dbus/dbus.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libproxy/CMakeFiles/config_pacrunner.dir/build.make:85: libproxy/CMakeFiles/config_pacrunner.dir/modules/config_pacrunner.cpp.o] Error 1
make[2]: Leaving directory '/tmp/libproxy-20201211-2743-1b34n46/libproxy-0.4.16/build'
make[1]: *** [CMakeFiles/Makefile2:1547: libproxy/CMakeFiles/config_pacrunner.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 96%] Linking CXX shared module config_kde.so
```

Even when I symlink the `.../include/dbus-0.1/dbus` into `.../include/dbus/`, the dependency is still broken:
```
[ 95%] Linking C executable proxy
cd /tmp/libproxy-20201211-4811-x63lou/libproxy-0.4.16/build/utils && /home/linuxbrew/.linuxbrew/Cellar/cmake/3.19.1/bin/cmake -E cmake_link_script CMakeFiles/proxy.dir/link.txt --verbose=1
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super/gcc-10 -DNDEBUG -rdynamic CMakeFiles/proxy.dir/proxy.c.o -o proxy  -Wl,-rpath,/tmp/libproxy-20201211-4811-x63lou/libproxy-0.4.16/build/libproxy: ../libproxy/libproxy.so.1.0.0
In file included from /tmp/libproxy-20201211-4811-x63lou/libproxy-0.4.16/libproxy/modules/config_pacrunner.cpp:24:
/home/linuxbrew/.linuxbrew/include/dbus/dbus.h:29:10: fatal error: dbus/dbus-arch-deps.h: No such file or directory
   29 | #include <dbus/dbus-arch-deps.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libproxy/CMakeFiles/config_pacrunner.dir/build.make:85: libproxy/CMakeFiles/config_pacrunner.dir/modules/config_pacrunner.cpp.o] Error 1
make[2]: Leaving directory '/tmp/libproxy-20201211-4811-x63lou/libproxy-0.4.16/build'
make[1]: *** [CMakeFiles/Makefile2:1547: libproxy/CMakeFiles/config_pacrunner.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/tmp/libproxy-20201211-4811-x63lou/libproxy-0.4.16/build'
[ 95%] Built target proxy
```

**Comment by danielnachun** [comment]
Right, I forgot I had to add that as well.  You'll also have to add `dbus` as a dependency.  Add `depends_on dbus` after line 16 and it should work.

**Comment by iMichka** [comment]
Fixed by #21777

**Comment by byxrolland** [comment]
@danielnachun It works now. Thanks!
